### PR TITLE
Vectorize local search

### PR DESCRIPTION
**Pull request by mfeurer**
This PR vectorizes the local search.

### CODE REVIEW

**Comment by mfeurer** [comment]
Okay, so I commented the code. Feedback welcome @mlindauer @AndreBiedenkapp @KEggensperger

**Comment by AndreBiedenkapp** [comment]
@mfeurer Code looks correct to me

File: smac/optimizer/ei_optimization.py [hunk 1]
```diff
@@ -247,75 +252,139 @@ def _get_initial_points(self, num_points, runhistory):
...
+                            if acq_val[acq_index] > acq_val_incumbents[i]:
+                                incumbents[i] = neighbors[acq_index]
+                                acq_val_incumbents[i] = acq_val[acq_index]
+                                new_neighborhood[i] = True
+                                improved[i] = True
+                                local_search_steps[i] += 1
+                                neighbors_w_equal_acq[i] = []
+                                obtain_n[i] = 1
+                            # Found an equally well performing configuration, keeping it for plateau walking
+                            elif acq_val[acq_index] == acq_val_incumbents[i]:
```
Comment (mlindauer) [inline smac/optimizer/ei_optimization.py]: I don't get why you distinguish between plateau steps and improving steps here.

Comment (mfeurer) [reply smac/optimizer/ei_optimization.py]: It was the same before, I did not change the behavior here.

**Comment by mfeurer** [comment]
Yeah, it's speed vs readability :(

I checked that the rosenbrock example is quantitatively similar. 

I also used the local search procedure to actually maximize (-1 * rosenbrock4d) from the unit tests, and the new versions actually performs better :open_mouth: no idea why...

**Comment by mfeurer** [comment]
I added an additional unit test to make sure that the local search procedure actually does something.

**Pull request merged**